R/subset_datarqst.R

```diff
@@ -30,73 +42,189 @@
+      dplyr::filter(.data$Species %in% c('CONFLICTED', 'CCCCCC')) %>%
+      dplyr::select(.data$IndvID)
+
+    #Progress bar
+    pb <- progress::progress_bar$new(total = length(unique(ConflSpec_IndvIDs$IndvID)),
+                                     format = "[:bar] :percent ~:eta remaining",
+                                     clear = FALSE)
 
-  save_path <- dirname(file)
+    #Determine which individuals were never identified as a species of interest
+    Irrelevant_IndvIDs <- output_capture %>%
+      dplyr::filter(.data$IndvID %in% ConflSpec_IndvIDs$IndvID) %>%
+      dplyr::arrange(.data$IndvID, .data$BreedingSeason, .data$CaptureDate, .data$CaptureTime) %>%
+      dplyr::group_by(.data$IndvID) %>%
+      dplyr::summarise(Relevant = purrr::map_chr(.x = list(unique(na.omit(.data$Species))),
+                                                 .f = ~{
+                                                 pb$tick()
+        if(any((..1) %in% SpeciesInt)){
 
-  utils::write.csv(x = output_brood, file = paste0(save_path, "\\Brood_data", as.numeric(Sys.Date()), ".csv"), row.names = F)
+          return("yes")
 
-  utils::write.csv(x = output_capture, file = paste0(save_path, "\\Capture_data", as.numeric(Sys.Date()), ".csv"), row.names = F)
+        } else {
 
-  utils::write.csv(x = output_individual, file = paste0(save_path, "\\Individual_data", as.numeric(Sys.Date()), ".csv"), row.names = F)
+          return("no")
 
-  utils::write.csv(x = output_location, file = paste0(save_path, "\\Location_data", as.numeric(Sys.Date()), ".csv"), row.names = F)
+        }
 
-  invisible(NULL)
+      }), .groups = "drop") %>%
+      dplyr::rowwise() %>%
+      #dplyr::ungroup() %>%
+      dplyr::filter(.data$Relevant == "no")
+
+    #Remove individuals never identified as one of the species of interest
+    output_individual <- output_individual %>%
+      filter(!(.data$IndvID %in% Irrelevant_IndvIDs$IndvID))
+
+    output_capture <- output_capture %>%
+      filter(!(.data$IndvID %in% Irrelevant_IndvIDs$IndvID))
+  }
+
+  #Combine output into one R object (and return it)
+  output_data <- list(Brood_data = output_brood,
+                      Capture_data = output_capture,
+                      Individual_data = output_individual,
+                      Location_data = output_location)
+
+  message(crayon::green("Data subset returned as R object."))
+  return(output_data)
+
+  if(save){
+
+    #Set save path
+
+    save_path <- paste0(dirname(file), "/", as.character(Sys.Date()), "_DataRequest")
+
+    if(!any(grepl(pattern = paste0(as.character(Sys.Date()), "_DataRequest"), x = list.dirs(dirname(file))))){
+
+      dir.create(save_path)
+
+    }
+
+
+    #Save output as RDS or .csv files
+    if(output_type == "R"){
+
+      saveRDS(output_data, file = paste0(save_path, "/Output_data_", as.character(Sys.Date()), ".RDS"))
+
+      message(crayon::cyan(crayon::bold("Data subset saved as R data file (.RDS).")))
+
+    }else{
```

Comment:
Besides the above issue, this will also save as csv with *any* other value of `output_type`, even e.g. if the user writes `output_type = 'r'`. I think we want to explicitly check that `output_type = 'csv'` before we save any .csv files.

```suggestion
}else if(output_type == "csv"){
```